### ISSUE TITLE
Make close and clear buttons more consistent across system

### ISSUE THREAD

**Issue by ashetland** [Esri/calcite-design-system#6640]
### Description

Close buttons across Calcite exhibit some inconsistencies due to differences in implementation. Ideally close buttons would be consistent in both design and implementation throughout the system.

### Acceptance Criteria
**Update close & clear button padding**
- Use `calcite-action` for close and clear buttons
- Replace any instance of `x-button` with `calcite-action` and sunset internal `x-button` component
- Update padding in consuming component 

### Relevant Info

Part of 4.0's Phase 2

Blocked by: #10759
Requires `transparent-inverse` colors tokens from #11565 

### Which Component

- `alert`
- `autocomplete`
- `chip`
- `combobox` / `input time zone`
- `inline editable`
- `input`
- `input-number`
- `input-text`
- `input-date-picker` (related issue #9440)
- `input-time-picker` (related issue #9440)
- `list item`
- `modal` (deprecated, may not need)
- `notice`
- `panel`, `flow`, `flow item`, `dialog`
- `popover`
- `tab title`
- `tip`, `tip manager` (deprecated, may not need)

### Priority impact

p4 - not time sensitive

### Esri team

Calcite (design)

**Comment by ashetland** [comment]
Need to rescope this to include Close and Clear buttons. Basic design of x-button is complete and should be ready to propagate in design files.

**Comment by jcfranco** [comment]
For posterity, the z-index workaround for #7272 will probably not be needed if tabs' closable button ends up being round.

**Comment by ashetland** [comment]
To do

- [ ] Verify above list and add any missing
- [ ] Determine which component should get round or square buttons (or if everything just goes round)
- [ ] Make visual catalogue of changes

**Comment by alisonailea** [comment]
I think code-wise we should create a mixin similar to the Floating UI functions that all components can share.

**Comment by driskull** [comment]
Currently, we only have an internal functional component to handle close buttons. We should consider using action or enhancing it to meet the needs of close buttons. It would be good to use an existing component if possible. Ideally, whatever we do we have public so other users can create componets with close buttons that match ours.

**Comment by ashetland** [comment]
I've wondered about using Action too. Action would need more scales added (or a way to make them smaller) to accommodate all the sizes we need. It would also need the ability to be round. For example, Chips use a round 16px close button. Maybe rounding it could be achieved without adding that to Actions? In any other context, round Actions could be weird?

**Comment by driskull** [comment]
Action or Button make sense from a close look but it could warrant its own component if those won't be able to handle it.

**Comment by ashetland** [comment]
Closing in favor of #10777. Consistent close and clear buttons are outlined there alongside other changes required from the updated Action design from #10759.
